src/ros_filter.cpp

```diff
@@ -3061,10 +3058,24 @@ namespace RobotLocalization
               }
               else
               {
-                // Has been initialized - use the alpha-beta filter
+                // Grab a local variable since it might be changed under the special case below.
+                double local_previous_yaw_offset = imuDynamicCorrectionData_[topicName].yaw_offset_;
+                if(::fabs(yaw_offset - imuDynamicCorrectionData_[topicName].yaw_offset_) > PI)
```

Comment:
That logic might have a bug which would occur because it is not calculating the shortest angular distance.
Consider the `ros::angles` [implementation](https://github.com/ros/angles/blob/f6b75c7c95b6c037ed0589678504c4a48082f20b/angles/include/angles/angles.h#L104) which then normalizes the angle.

Is there a reason to not use the shortest angular distance?

Reply:
I might be misunderstanding, but this logic is not supposed to calculate the shortest angular distance, since abs(shortest angular distance) would always be <= PI. The goal of this line is to determine if there is a wrap occurring across the -PI/PI boundary to properly handle the alpha-beta filter. I do think there is an alternate implementation, though. Instead of new=previous*alpha + current_measurement*(1-alpha), I can do new=(previous-current_measurement)*alpha + current_measurement. Previous-current_measurement now can use shortest angular distance. Less traditional formulation, but works as well, as long as the function being used handles the signs correctly. Will test this out.